### PR TITLE
Remove cuda cache emptying in evals.py

### PR DESCRIPTION
**Pull request by tomMcGrath**
# Description

Currently evals.py includes a call to `torch.cuda.empty_cache()`. This causes inconsistent memory usage across runs, which makes OOMs unpredictable and might cause extra recompilation (though I'm not sure on this). `torch.cuda.empty_cache` doesn't actually free up additional memory for pytorch - it simply releases it from the pytorch allocator (I was surprised by this, discussion [here](https://pytorch.org/docs/stable/notes/cuda.html#memory-management) and [here](https://discuss.pytorch.org/t/about-torch-cuda-empty-cache/34232)). It's also not recommended for use by end-users except in edge cases, so I propose we remove it.

I've tested this change with and without compilation on GPT-2-S SAEs. All unit tests continue to pass.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] I have commented my code, particularly in hard-to-understand areas (n/a: code removal)
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have not rewritten tests relating to key interfaces which would affect backward compatibility

### You have tested formatting, typing and unit tests (acceptance tests not currently in use)

- [x] I have run `make check-ci` to check format and linting. (you can run `make format` to format code if needed.)

### Performance Check.

If you have implemented a training change, please indicate precisely how performance changes with respect to the following metrics:
- [ ] L0
- [ ] CE Loss
- [ ] MSE Loss
- [ ] Feature Dashboard Interpretability

Please links to wandb dashboards with a control and test group.

### CODE REVIEW

**Comment by jbloomAus** [comment]
Huh, weird! I didn't know this. Thanks for catching that.

**Comment by jbloomAus** [comment]
Accepting when builds pass.

**Comment by tomMcGrath** [comment]
Yeah it's a bit weird! I'm sending over a summary about VRAM usage in a min

**Comment by hijohnnylin** [comment]
sorry for my lack of knowledge on this, does this mean that we should remove all usages of empty_cache since it's not recommended for end users?
for example in these scripts:
https://github.com/jbloomAus/SAELens/blob/bdef2cf0e4ad3e3070f9be1c46a1adf13094eb13/scripts/caching_replication_how_train_saes.py#L52-L55
